cluster-autoscaler/cloudprovider/azure/azure_config.go

```diff
@@ -87,8 +87,16 @@ type Config struct {
 	Location       string `json:"location" yaml:"location"`
 	TenantID       string `json:"tenantId" yaml:"tenantId"`
 	SubscriptionID string `json:"subscriptionId" yaml:"subscriptionId"`
-	ResourceGroup  string `json:"resourceGroup" yaml:"resourceGroup"`
-	VMType         string `json:"vmType" yaml:"vmType"`
+	ClusterName    string `json:"clusterName" yaml:"clusterName"`
+	// ResourceGroup is the MC_ resource group where the nodes are located.
+	ResourceGroup string `json:"resourceGroup" yaml:"resourceGroup"`
+	// ClusterResourceGroup is the resource group where the cluster is located.
+	ClusterResourceGroup string `json:"clusterResourceGroup" yaml:"clusterResourceGroup"`
+	VMType               string `json:"vmType" yaml:"vmType"`
+
+	// ARMBaseURL is the URL to use for operations for the VMs pool.
+	// It can override the default public ARM endpoint for VMs pool scale operations.
+	ARMBaseURL string `json:"armBaseURL" yaml:"armBaseURL"`
```

Comment:
This is used only by AgentPool client, which may not be obvious to those reading the code (especially since cloud-provider client configuration supports overriding RM endpoint for non-public clouds). Should we reflect this in the field name? Maybe something with "AKS" and/or "AP" in the name?

Related: At some point, please document the new configuration options in cloudprovider/azure/README.md

Reply:
This variable prefix with "ARM" which indicates **Azure** Resource Manager, so I don't think we need to prefix with AKS/AP here if that makes sense

Reply:
All existing clients talk to ARM, this field affects only AKS AgentPool client. Ideally the intended use of the variable should be clear from its name. Comments are bonus, but don't replace unambiguous naming.

Reply:
Updated the name to `ARMBaseURLForAPClient`